@memZono/map.m

```diff
@@ -0,0 +1,78 @@
+function out = map(obj1,obj2,inDims,outDims)
+    % check to ensure only 
+    memZonoBools = [isa(obj1,'memZono'),isa(obj2,'memZono')];
+    if all(memZonoBools)
+        error('Map currently not yet implimented for functional mapping');
+    else
+        if memZonoBools(1)
+            in = obj1;
+            M = obj2;
+        elseif memZonoBools(2)
+            in = obj2;
+            M = obj1;
+        else
+            error('this should never be reached');
+        end
+        out = linMap(in,M,inDims,outDims);
+    end
+
+    % switch memZonoBools
+    %     case [1 1]
+    %         error('Map currently not yet implimented for functional mapping');
+    %     case [1 0]
+    %         in = obj1;
+    %         M = obj2;
+    %         out = linMap(in,M,inDims,outDims);
+    %     case [0 1]
+    %         in = obj2;
+    %         M = obj1;
+    %         out = linMap(in,M,inDims,outDims);
+    %     case [0 0]
+    %         error("this shouldn't be possible")
+    % end
+end
+
+% Local linMap... since not a method currently
+function out = linMap(in,M,inDims,outDims);
```

Comment:
Seems simpler to move code from linMap and put it into map directly...
```suggestion
function out = linMap(in,M,inDims,outDims);
```

Reply:
This is nessicary for checks and then the eventual implementation of functional mapping (which is an a minimum a future release tasks if not ideally now)